addons/binding/org.openhab.binding.mcp23017/src/main/java/org/openhab/binding/mcp23017/handler/AbstractMcp23017Handler.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mcp23017.handler;
+
+import com.pi4j.gpio.extension.mcp.MCP23017GpioProvider;
+import com.pi4j.gpio.extension.mcp.MCP23017Pin;
+import com.pi4j.io.gpio.Pin;
+import com.pi4j.io.i2c.I2CFactory.UnsupportedBusNumberException;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.openhab.binding.mcp23017.Mcp23017BindingConstants.ADDRESS;
+import static org.openhab.binding.mcp23017.Mcp23017BindingConstants.BUS_NUMBER;
+import static org.openhab.binding.mcp23017.Mcp23017BindingConstants.PIN;
+
+/**
+ * The {@link AbstractMcp23017Handler} is base class for MCP23017 chip support
+ *
+ * @author Anatol Ogorek
+ */
+public abstract class AbstractMcp23017Handler extends BaseThingHandler {
+
+    protected static final Map<Integer, MCP23017GpioProvider> MCP_PROVIDERS = new HashMap<>();
+    protected Integer address;
+    protected Integer busNumber;
+    protected Pin pin;
+
+    /**
+     * the polling interval mcp23071 check interrupt register (optional, defaults to 50ms)
+     */
+    protected int pollingInterval = 50;
```

Comment:
This sounds / looks like a constant: `protected static final POLLING_INTERVAL` ?